File: CS/UseLegacyReportWizard/Form1.cs
using DevExpress.DataAccess.UI.Native.Sql;
using DevExpress.DataAccess.Wizard;
using System;

namespace UseLegacyReportWizard {
    public partial class Form1 : System.Windows.Forms.Form {
        public Form1() {
            InitializeComponent();

            // Replace the default ISqlWizardOptionsProvider service with a custom one.
            reportDesigner1.RemoveService(typeof(ISqlWizardOptionsProvider));
            reportDesigner1.AddService(typeof(ISqlWizardOptionsProvider), new MySqlWizardOptionsProvider(() =>
                reportDesigner1.DataSourceWizardSettings.SqlWizardSettings.ToSqlWizardOptions()));
        }

        // Implement a custom SqlWizardOptionsProvider and specify the SqlWizardOptions.
        public class MySqlWizardOptionsProvider : ISqlWizardOptionsProvider {
            readonly Func<SqlWizardOptions> getOptions;

            public MySqlWizardOptionsProvider(Func<SqlWizardOptions> getOptions) {
                this.getOptions = getOptions;
            }
            SqlWizardOptions ISqlWizardOptionsProvider.SqlWizardOptions {
                get { return getOptions() & SqlWizardOptions.MultiQueryWizard; }
            }
        }
    }
}

File: VB/UseLegacyReportWizardVB/Form1.vb
Imports DevExpress.DataAccess.UI.Native.Sql
Imports DevExpress.DataAccess.Wizard
' ... 

Public Class Form1
    Inherits System.Windows.Forms.Form
    Public Sub New()
        InitializeComponent()

        ReportDesigner1.RemoveService(GetType(ISqlWizardOptionsProvider))
        ReportDesigner1.AddService(GetType(ISqlWizardOptionsProvider), New MySqlWizardOptionsProvider(Function() _
                ReportDesigner1.DataSourceWizardSettings.SqlWizardSettings.ToSqlWizardOptions()))
    End Sub
End Class

Public Class MySqlWizardOptionsProvider
    Implements ISqlWizardOptionsProvider
    ReadOnly getOptions As Func(Of SqlWizardOptions)

    Public Sub New(getOptions As Func(Of SqlWizardOptions))
        Me.getOptions = getOptions
    End Sub
    Private ReadOnly Property ISqlWizardOptionsProvider_SqlWizardOptions() As SqlWizardOptions Implements ISqlWizardOptionsProvider.SqlWizardOptions
        Get
            Return getOptions() And SqlWizardOptions.MultiQueryWizard
        End Get
    End Property
End Class